uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Form/CreateUserRecordTaskForm.php

```diff
@@ -44,13 +44,12 @@ public function postProcess() {
     $contactsToCreate = $this->getValidContactsForCreation();
 
     foreach ($contactsToCreate as $contact) {
-      $user = $this->drupalUserService->findUserByEmail($contact['email']);
-      if (!$user) {
+      $userExists = $this->drupalUserService->userExistsForEmail($contact['email']);
+      if (!$userExists) {
         $this->createAccount($contact, $roles);
       }
       else {
-        $this->drupalUserService->processUserAccount($user, $roles);
-        $this->addUfId($user, $contact['id']);
+        $this->drupalUserService->restoreAccount($contact['email'], $roles, $contact['id']);
```

Comment:
We actually don't need a restoreAccount method in the DrupalUserService. The Service should not be concerned about setting Ufid for civicrm, its not its responsibility. What I meant earlier was to retain the addUFId method in the `CreateUserRecordTaskForm` class and create a new method called restoreAccount in this class.

```php
private function restoreAccount($contact, roles) {
$this->drupalUserService->addRoles($contact['email], $roles);
$this->addUfId($contact);
}
```
So the only new method we'll be adding to the DrupalUserService will be the `userExistsForEmail` method.